features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/wrappers/CanvasWrapper.kt

```diff
@@ -13,8 +13,18 @@ import com.datadog.android.api.InternalLogger
 internal class CanvasWrapper(
     private val logger: InternalLogger = InternalLogger.UNBOUND
 ) {
+
     internal fun createCanvas(bitmap: Bitmap): Canvas? {
-        @Suppress("SwallowedException", "TooGenericExceptionCaught")
+        if (bitmap.isRecycled || !bitmap.isMutable) {
+            logger.log(
+                level = InternalLogger.Level.ERROR,
+                target = InternalLogger.Target.MAINTAINER,
+                { INVALID_BITMAP }
+            )
+            return null
+        }
+
+        @Suppress("TooGenericExceptionCaught")
         return try {
             Canvas(bitmap)
         } catch (e: IllegalStateException) {
```

Comment:
minor: the comment on the line below should be updated to `should never happen since we are passing a mutable bitmap`